### ISSUE TITLE
wrong error message when popping an empty stash

### ISSUE THREAD

**Issue by kevgo** [git-town/git-town#2227]
### steps to reproduce:

- run a Git Town command that stops with an error and asks you to fix it and then run `git town continue`
- fix the error and clear the Git stash
- run `git town continue`

### expected result

Git Town ignores the issue that it cannot pop the Git stash because there is no Git stash.

### actual result

Git Town thinks its inability to pop the stash is a failure condition and prints this error:

```
(other Git Town output here...)

[main] git stash pop
No stash entries found.

Error: conflicts between your uncommmitted changes and the main branch

To abort, run "git-town abort".
To continue after having resolved conflicts, run "git-town continue".
To continue by skipping the current branch, run "git-town skip".
```

**Comment by IGassmann** [comment]
Ever since I upgraded to version 10, I encounter this issue every time I execute ﻿`git sync` on an up-to-date branch and with an empty stash:

```shell
❯ git sync

[main] git fetch --prune --tags

[main] git add -A

[main] git stash
No local changes to save

[main] git rebase origin/main
Current branch main is up to date.

[main] git push --tags
Everything up-to-date

[main] git stash pop
No stash entries found.


Error: conflicts between your uncommmitted changes and the main branch

To abort, run "git-town abort".
To continue after having resolved conflicts, run "git-town continue".
To continue by skipping the current branch, run "git-town skip".
```

I attempted to run ﻿`git town status reset` in an effort to resolve the issue, but it did not resolve it.

**Comment by IGassmann** [comment]
The issue is also occurring with the the `git hack` command:

```shell
❯ git hack new-branch

[main] git fetch --prune --tags

[main] git add -A

[main] git stash
No local changes to save

[main] git rebase origin/main
Current branch main is up to date.

[main] git branch new-branch main

[main] git checkout new-branch
Switched to branch 'new-branch'

[new-branch] git stash pop
No stash entries found.


Error: conflicts between your uncommmitted changes and the main branch

To abort, run "git-town abort".
To continue after having resolved conflicts, run "git-town continue".
```

**Comment by kevgo** [comment]
Thanks for the bug report, interesting edge case! It seems the problem you experience happens because Git Town wrongfully assumes your branch contains uncommitted changes. It therefore runs `git stash` (which in your case stashes nothing) and then `git stash pop` (which in your case fails because there is nothing to unstash, resulting in the misleading error message).

Let's find out why this happens. @IGassmann can you please post the output of this command:

```
git sync --verbose
```

Please make sure to scrub any confidential information like your GitHub tokens from the output. It would show up in the beginning when Git Town runs ` git config -lz`. Thanks!

**Comment by IGassmann** [comment]
@kevgo the issue is happening with all commands actually whenever the stash is empty.

Here is the verbose output:

```shell
❯ git sync --verbose

(verbose) git version
git version 2.42.1

(verbose) git config -lz --global
[redacted]

(verbose) git config -lz --local
[redacted]

(verbose) git stash list

(verbose) git branch -vva
* main                                                                     bb97914f [origin/main] Commit message

(verbose) git remote
origin

[main] git fetch --prune --tags

(verbose) git branch -vva
  * main                                                                     bb97914f [origin/main] Commit message

(verbose) git rev-parse --verify --abbrev-ref @{-1}
igor/previous-branch

(verbose) git status --ignore-submodules
## main...origin/main

[main] git add -A

[main] git stash
No local changes to save

[main] git rebase origin/main
Current branch main is up to date.

(verbose) git rev-list --left-right main...origin/main

(verbose) git show-ref --quiet refs/heads/main

[main] git push --tags
Everything up-to-date

(verbose) git show-ref --quiet refs/heads/igor/previous-branch

(verbose) git rev-parse --verify --abbrev-ref @{-1}
igor/previous-branch

[main] git stash pop
No stash entries found.

(verbose) git config -lz --global
[redacted]
(verbose) git config -lz --local
[redacted]
(verbose) git branch -vva
  * main                                                                     bb97914f [origin/main] Commit message

(verbose) git stash list

(verbose) git status --ignore-submodules
## main...origin/main

Ran 25 shell commands.
Error: conflicts between your uncommmitted changes and the main branch

To abort, run "git-town abort".
To continue after having resolved conflicts, run "git-town continue".
To continue by skipping the current branch, run "git-town skip".
```

**Comment by IGassmann** [comment]
I've recorded a video that shows the issue is happening across all commands. It seems that the problem is that commands always try to unstash changes even when they shouldn't.


https://github.com/git-town/git-town/assets/4291707/88b78e6f-2bbe-41d5-be20-7fe3635d6806

My git version: `2.42.1`

**Comment by kevgo** [comment]
Thanks for the details! The video has also been helpful. I notice that the output of `git status --ignore-submodules` on your machine differs from my machine. When I run this command, I get this:

```
~/d/git-town [main] > git status --ignore-submodules                                                                                                  (base) 
On branch main
Your branch is up to date with 'origin/main'.

Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
	modified:   src/git/backend_commands.go
	modified:   src/git/backend_commands_test.go

no changes added to commit (use "git add" and/or "git commit -a")
```

When you run the exact same command, it seems you get just something like this:

```
## main...origin/main
```

I suspect you enabled  the `status.short` and `status.branch` config settings of Git. Can you please confirm?

**Comment by kevgo** [comment]
I have released [v10.0.2](https://github.com/git-town/git-town/releases/tag/v10.0.2) with a fix for the problem of getting a wrong error message when short output is enabled for Git. Please take it for a ride and let me know if the problem still exists. Thanks!

**Comment by IGassmann** [comment]
> I suspect you enabled the `status.short` and `status.branch` config settings of Git. Can you please confirm?

That's right. I have both enabled.

> I have released [v10.0.2](https://github.com/git-town/git-town/releases/tag/v10.0.2) with a fix for the problem of getting a wrong error message when short output is enabled for Git. Please take it for a ride and let me know if the problem still exists. Thanks!

That fixed it! Thanks :)
